src/include/hip/rpp_hip_common.hpp

```diff
@@ -1424,9 +1470,109 @@ __device__ __forceinline__ void rpp_hip_math_subtract24_const(d_float24 *src_f24
     dst_f24->f4[5] = src_f24->f4[5] - subtrahend_f4;
 }
 
+// d_float8 multiply
+
+__device__ __forceinline__ void rpp_hip_math_multiply8(d_float8 *src1Ptr_f8, d_float8 *src2Ptr_f8, d_float8 *dstPtr_f8)
+{
+    dstPtr_f8->f4[0] = src1Ptr_f8->f4[0] * src2Ptr_f8->f4[0];
+    dstPtr_f8->f4[1] = src1Ptr_f8->f4[1] * src2Ptr_f8->f4[1];
+}
+
+// d_float24 multiply
+
+__device__ __forceinline__ void rpp_hip_math_multiply24(d_float24 *src1Ptr_f24, d_float24 *src2Ptr_f24, d_float24 *dstPtr_f24)
+{
+    dstPtr_f24->f4[0] = src1Ptr_f24->f4[0] * src2Ptr_f24->f4[0];
+    dstPtr_f24->f4[1] = src1Ptr_f24->f4[1] * src2Ptr_f24->f4[1];
+    dstPtr_f24->f4[2] = src1Ptr_f24->f4[2] * src2Ptr_f24->f4[2];
+    dstPtr_f24->f4[3] = src1Ptr_f24->f4[3] * src2Ptr_f24->f4[3];
+    dstPtr_f24->f4[4] = src1Ptr_f24->f4[4] * src2Ptr_f24->f4[4];
+    dstPtr_f24->f4[5] = src1Ptr_f24->f4[5] * src2Ptr_f24->f4[5];
+}
+
+// d_float8 multiply with constant
+
+__device__ __forceinline__ void rpp_hip_math_multiply8_const(d_float8 *src_f8, d_float8 *dst_f8, float4 multiplier_f4)
+{
+    dst_f8->f4[0] = src_f8->f4[0] * multiplier_f4;
+    dst_f8->f4[1] = src_f8->f4[1] * multiplier_f4;
+}
+
+// d_float24 multiply with constant
+
+__device__ __forceinline__ void rpp_hip_math_multiply24_const(d_float24 *src_f24, d_float24 *dst_f24, float4 multiplier_f4)
+{
+    dst_f24->f4[0] = src_f24->f4[0] * multiplier_f4;
+    dst_f24->f4[1] = src_f24->f4[1] * multiplier_f4;
+    dst_f24->f4[2] = src_f24->f4[2] * multiplier_f4;
+    dst_f24->f4[3] = src_f24->f4[3] * multiplier_f4;
+    dst_f24->f4[4] = src_f24->f4[4] * multiplier_f4;
+    dst_f24->f4[5] = src_f24->f4[5] * multiplier_f4;
+}
+
+__device__ __forceinline__ float rpp_hip_math_inverse_sqrt1(float x)
+{
+    float xHalf = 0.5f * x;
+    int i = *(int*)&x;                              // float bits in int
```

Comment:
please use the builtin hip functions to convert float to int (__float2int_rn)

Reply:
Actually as in the above comments, this inverse_sqrt algorithm needs the same float bits (same order of sign-exponent-mantissa in IEEE format) but to be interpreted as int for the right shifting. But the __float2int_rn() may end up converting the value itself to integer.

Reply:
@asalmanp: It is not really converting from float to int. It is just representing the exact float number as int.
@r-abishek Please add link to the paper for the algorithm